### ISSUE TITLE
[Bug]: react/boolean-prop-naming validateNested config doesn't work with typescript interfaces

### ISSUE THREAD

**Issue by Phebonacci** [jsx-eslint/eslint-plugin-react#3856]
### Is there an existing issue for this?

- [x] I have searched the existing issues and my issue is unique
- [x] My issue appears in the command-line and not only in the text editor

### Description Overview

The `validateNested` configuration option doesn't work for typescript interface props.

Sample code:
```
import React from 'react';

type ButtonProps = {
    children: React.ReactNode;
    foo: { bar: boolean };
    baz: boolean;
}

const Button = ({ children }: ButtonProps): JSX.Element => {
    return <button type="button">{children}</button>;
};

export default Button;
```

Link to sample repo: https://github.com/Phebonacci/sample-nextra

The `react/boolean-prop-naming` rule doesn't trigger for nested boolean props. I verified my setup by running eslint against a sample file with a top level boolean prop and a nested boolean prop. As can be seen on the screenshot below, the rule triggers for the top level boolean prop, however, it doesn't trigger for the nested boolean prop even with `validateNested` config set to true:
![Image](https://github.com/user-attachments/assets/305f90aa-b909-4a27-839c-9c70bf8ef116)

To reproduce the issue, run:
```
yarn eslint src/components/Button.tsx
```

### Expected Behavior

The `react/boolean-prop-naming` rule should also trigger for the nested boolean props.

### eslint-plugin-react version

v7.37.2

### eslint version

v8.57.1

### node version

v18.19.0

**Comment by ljharb** [comment]
`type` and `interface` are different things - i assume it fails to work with either?
